_sources/functions_in_cpp/Exercises.rst

```diff
@@ -0,0 +1,314 @@
+
+        -   ``22 % 2 == 0``, so you enter the ``if`` block, not the else.
+
+    -   ``He who laughs at himself never runs out of things to laugh at.``
+
+        -   ``'b'`` is less than ``'m'``, so you would enter the ``if`` block, not the ``else if``.
+
+    -   ``You will be hungry again in one hour.``
+
+        -   ``'b'`` is less than ``'m'``, so you would enter the ``if`` block, not the ``else if``.
+
+    -   ``Fortune not found? Abort, retry, ignore.``
+
+        -   ``'b'`` is less than ``'m'``, so you would enter the ``if`` block, not the ``else``.
+
+.. mchoice:: functions_in_cpp_mcq_7
+    :practice: T
+
+    Are there any issues with the code below?
+
+    .. code-block:: cpp
+
+      bool isEven (int num) {
+        if (num % 2 == 0) {
+          return true;
+        }
+      }
+
+    - Yes, we have to return either 0 or 1.
+
+      - Returning a 0 or 1 would be returning an ``int``, even though booleans evaluate to 0 or 1.
+
+    - Yes, we cannot pass an ``int`` into a ``bool`` function.
+
+      - The type of variables in the parameter list do not affect the return type.
+
+    - Yes, there is no case for odd numbers.
+
+      + Since we never established an else clause, if the input was an odd number, the function would not return anything despite not being a void function.
+
+    - There are no issues with the code.
+
+      - There is an issue with the code. Can you find it?
+
+.. mchoice:: functions_in_cpp_mcq_8
+    :practice: T
+
+    What is the value of ``x`` after the code is run?
+
+    .. code-block:: cpp
+
+      void double(int num) {
+        num = 2 * num;
+      }
+
+      int main() {
+        int x = 4;
+        double(x);
+        double(x);
+      }
+
+    - 16
+
+      - Although that might be what the ``double`` function hopes to achieve, is that what the value of ``x`` is?
+
+    - 8
+
+      - Take a closer look at what ``double`` does.
+
+    - 4
+
+      + Since ``num`` is not pass by reference, ``x`` is not actually modified.
+
+    - 2
+
+      - Take another look at the program.
+
+.. mchoice:: functions_in_cpp_mcq_9
+    :practice: T
+
+    Are there any issues with the code below?
+
+    .. code-block:: cpp
+
+      void moonWeight (double earth) {
+        double moon = 0.165 * earth;
+        cout << "You would weigh " << moon << " pounds on the moon." << endl;
+        return moon;
+      }
+
+    - Yes, we cannot have ``cout`` statements in a function.
+
+      - We are allowed to use ``cout`` statements in a function.
+
+    - Yes, we cannot return anything.
+
+      + ``void`` functions do not have return values, so we cannot return ``moon``.
+
+    - Yes, we need to return the output statement.
+
+      - ``void`` functions do not have return values.
+
+    - There are no issues with the code.
+
+      - There is an issue with the code. Can you find it?
+
+.. mchoice:: functions_in_cpp_mcq_10
+    :practice: T
+
+    What is the return type of main?
```

Comment:
We don't really talk about this, so I think it would be better to remove this question.